R/file_cache.R

```diff
@@ -0,0 +1,542 @@
+#' Create a file cache object
+#'
+#' This creates a file cache which is to be used by sass for caching generated
+#' .css files.
+#'
+#' @param dir The directory in which to store the cached files. If `NULL` (the
+#'   default), then it will be set to a subdirectory of the system temp
+#'   directory named `R-sass-cache-username`. Because this cache directory is
+#'   not in the R process's temp directory, it will persist longer than the R
+#'   process, typically until a system reboot. Additionally, it will be shared
+#'   across R processes for the same user on the same system.
+#' @param max_size The maximum size of the cache. If the cache grows past this
+#'   size, the least-recently-used objects will be removed until it fits within
+#'   this size.
+#'
+#' @seealso sass_get_default_cache
+#'
+#' @examples
+#' \dontrun{
+#' # Create a cache with the default settings
+#' cache <- sass_file_cache()
+#'
+#' # Clear the cache
+#' cache$reset()
+#' }
+#'
+#' @export
+sass_file_cache <- function(dir = NULL, max_size = 40 * 1024 ^ 2) {
+  if (is.null(dir)) {
+    dir <- file.path(dirname(tempdir()), paste0("R-sass-cache-", Sys.info()[["user"]]))
+  }
+  FileCache$new(dir, max_size)
```

Comment:
Is there any reason why this isn't something like `FileCache$new(dir, ...)`?

Reply:
I just wanted to limit the external API, at least for now.